### PR TITLE
Swift 5

### PR DESCRIPTION
**Pull request by stephencelis**
Swift 5-compatible Prelude.

### CODE REVIEW

File: Sources/Either/Either.swift [hunk 1]
```diff
@@ -54,7 +54,7 @@ public func hush<L, R>(_ lr: Either<L, R>) -> R? {
   return lr.either(const(.none), R?.some)
 }
 
-public extension Either where L == Error {
```
Comment (stephencelis) [inline Sources/Either/Either.swift]: New warning. I guess a `public` extension automatically makes all members public now?

File: Sources/Frp/UIKit.swift [hunk 2]
```diff
@@ -99,7 +99,7 @@
 
   public let animationFrame: Event<()> = {
     let (event, push) = Event<()>.create()
-    _ = Animator(push)
+    _ = Animator { push(()) }
```
Comment (stephencelis) [inline Sources/Frp/UIKit.swift]: Surprised this tuple bit of code worked...though maybe we just haven't compiled for iOS in awhile...

File: Sources/Prelude/Collection.swift [hunk 3]
```diff
@@ -1,3 +1,47 @@
+public func drop<C: Collection>(while p: @escaping (C.Element) -> Bool) -> (C) -> C.SubSequence {
```
Comment (stephencelis) [inline Sources/Prelude/Collection.swift]: `Sequence` no longer has `SubSequence`.

File: Sources/Prelude/Comparable.swift [hunk 4]
```diff
@@ -20,12 +20,6 @@ extension Bool: Comparable {
   }
 }
 
-extension Never: Comparable {
```
Comment (stephencelis) [inline Sources/Prelude/Comparable.swift]: Free conformance these days. Gets rid of a warning.

File: Sources/Prelude/Equatable.swift [hunk 5]
```diff
@@ -1,9 +1,3 @@
-extension Never: Equatable {
```
Comment (stephencelis) [inline Sources/Prelude/Equatable.swift]: Another free conformance.

File: Sources/Prelude/Optional.swift [hunk 6]
```diff
@@ -7,7 +7,7 @@ public func optional<A, B>(_ default: @autoclosure @escaping () -> B) -> (@escap
 }
 
 public func coalesce<A>(with default: @autoclosure @escaping () -> A) -> (A?) -> A {
-  return optional(`default`) <| id
+  return optional(`default`()) <| id
```
Comment (stephencelis) [inline Sources/Prelude/Optional.swift]: Looks like auto-closures can no longer be implicitly passed to other functions that take auto-closures without an explicit call.

**Comment by ilyapuchka** [comment]
Any update on this?
I was trying to bring into the project a package of mine that depends on Prelude, but as it depends on latest versions of Tagged and snapshot-testing libraries it brings with it its development dependencies like XcodeGen, which are definitely not needed in the project =/
If this is blocked is it possible to bump just the package?

File: .swift-version [hunk 7]
```diff
@@ -1 +1 @@
-4.2.1
+swift-5.0-DEVELOPMENT-SNAPSHOT-2019-03-06-a
```
Comment (mbrandonw) [inline .swift-version]: we can use 5.0.2 right?

File: Dockerfile [hunk 8]
```diff
@@ -1,4 +1,4 @@
-FROM norionomura/swift:421
+FROM norionomura/swift:swift-5.0-branch
```
Comment (mbrandonw) [inline Dockerfile]: do we wanna switch to the official swift 5.0.2 docker image?

**Pull request merged**